module/zfs/zio.c

```diff
@@ -4465,9 +4469,18 @@ zio_done(zio_t *zio)
 		 * device is currently unavailable.
 		 */
 		if (zio->io_error != ECKSUM && zio->io_vd != NULL &&
-		    !vdev_is_dead(zio->io_vd))
+		    !vdev_is_dead(zio->io_vd)) {
```

Comment:
I believe we should add `&& !(zio->io_flags & ZIO_FLAG_IO_RETRY)` to this conditional so we don't over count error due to internal retries.  A similar check was part of `vdev_stat_update()`.